### ISSUE TITLE
value(..., const char*) should take std::string_view ?

### ISSUE THREAD

**Issue by Talkless** [nlohmann/std_json#16]
I see there are these signatures mentioned in the [current proposal](https://github.com/nlohmann/std_json/blob/640d731195d35c438b0ddd507b4b181eaf531f89/proposal.md#object-element-access-by-key-with-default-value):

```
string_type value(const typename object_type::key_type &, const char * default_value) const;
```
```
string_type value(const json_pointer &, const char * default_value) const;
```

Couldn't these take `string_view` instead?

There is a proposal [p0506r2](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0506r2.pdf) to change `string` and `const char*` into `string_view`, maybe it's fit here too.

**Comment by mariokonrad** [comment]
Yes, this is on my todo list. Niels' library is based on C++11, which is perfectly fine. For the proposal, which will be based on C++17 (at least), it certainly makes sense.
I just did not find the time during the last months to work on it much, but will continue in the (near) future.
